### PR TITLE
Module 5

### PR DESCRIPTION
**Pull request by MarcCharbo**

### CODE REVIEW

File: students/marc_charbo/lesson04/mailroom_v2.py [hunk 1]
```diff
@@ -85,9 +92,14 @@ def run():
 
 def main():
     try:
+        logging.basicConfig(filename='myapp.log', level=logging.INFO)
```
Comment (mdgregor) [inline students/marc_charbo/lesson04/mailroom_v2.py]: This should be defined at the top of your program, not within `main()`

File: students/marc_charbo/lesson05/mailroom_v3.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    print ('Please select one of three options: ')
+    print ('1- Send a Thank You')
+    print ('2- Create a Report')
+    print('3- Send letters to everyone')
+    print ('4- Quit')
+    selection = input('Enter your selection: ')
+    
+    return int(selection)
+
+dispatch_dict = {1:send_thank_you,2:create_report,3:send_letter,4:quit}
```
Comment (mdgregor) [inline students/marc_charbo/lesson05/mailroom_v3.py]: add some spacing here:
```
{1: send_thank_you, 2: create_report, 3: send_letter, 4: quit}
```

File: students/marc_charbo/lesson05/mailroom_v3.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    for key, value in DONOR_LIST.items():
+        file = open(key+'.txt', 'w')
+        file.write('Dear {},\n'.format(key))
+        file.write('\n')
+        file.write('\tThank you for your very kind donation of ${:.2f},\n'.format(sum(value)))
+        file.write('\n')
+        file.write('\tIt will be put to very good use.\n')
+        file.write('\n')
+        file.write('\t\t\tSincerely,\n')
+        file.write('\t\t\t\t-The Team')
```
Comment (mdgregor) [inline students/marc_charbo/lesson05/mailroom_v3.py]: better to define all this as a single string (using triple quotes so you don't need all the `\n` and `\t` )and then write to file

File: students/marc_charbo/lesson05/mailroom_v3.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+logging.config.fileConfig('logging.conf')
+# create logger
+logger = logging.getLogger('mailRoom')
+
+DONOR_LIST = {'Jim':[25.00,150.00,2000.00,100000.00],'Linda':[10000.25],'Bob':[5.03,100.01,6.00]}
+
+def send_letter():
+    """write thank you note to all users in donor list"""
+    for key, value in DONOR_LIST.items():
+        file = open(key+'.txt', 'w')
```
Comment (mdgregor) [inline students/marc_charbo/lesson05/mailroom_v3.py]: try using a `with` context manager here

File: students/marc_charbo/lesson04/mailroom_v2.py (hunk 1 above)
Comment (MarcCharbo) [reply students/marc_charbo/lesson04/mailroom_v2.py]: taking care of in v3

File: students/marc_charbo/lesson05/mailroom_v3.py (hunk 2 above)
Comment (MarcCharbo) [reply students/marc_charbo/lesson05/mailroom_v3.py]: done in v4. Thanks for feedback!

**Comment by MarcCharbo** [comment]
ok. i really need to this more everywhere

File: students/marc_charbo/lesson05/mailroom_v3.py (hunk 4 above)
Comment (MarcCharbo) [reply students/marc_charbo/lesson05/mailroom_v3.py]: @mdgregor could you elaborate on using with? thanks

**Pull request merged**